File: Data Structures & Algorithms/sort-colors/submission-1.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero = 0;
        one = 0;
        two = 0;

        for i in nums:
            if i == 0:
                zero += 1
            elif i == 1:
                one += 1
            else:
                two += 1
        
        i = 0
        while zero > 0:
            nums[i] = 0
            i += 1
            zero -= 1
        
        while one > 0:
            nums[i] = 1
            i += 1
            one -= 1

        while two > 0:
            nums[i] = 2
            i += 1
            two -= 1
        

File: Data Structures & Algorithms/rotate-array/submission-1.java
class Solution {
    public void rotate(int[] nums, int k) {
        int rotate = k;
        while(rotate >= nums.length){
            rotate -= nums.length;
        }

        if(rotate == 0){
            return;
        }
        int[] result = new int[nums.length];
        for(int i=0; i<nums.length; i++){
            int rot_counter = (i + k) % nums.length;
            result[rot_counter] = nums[i];
        }

        for(int i=0; i<nums.length; i++){
            nums[i] = result[i];
        }
    }
}